src/System.Private.CoreLib/shared/System/IO/FileStream.Windows.cs

```diff
@@ -61,7 +61,14 @@ private void Init(FileMode mode, FileShare share)
             int fileType = Interop.Kernel32.GetFileType(_fileHandle);
             if (fileType != Interop.Kernel32.FileTypes.FILE_TYPE_DISK)
             {
+                int errorCode = Marshal.GetLastWin32Error();
```

Comment:
To be clearer about the need for last error, don't get the error unless you have FILE_TYPE_UNKNOWN.  Something like:

``` C#
int errorCode = filetype == Interop.Kernel32.FileTypes.FILE_TYPE_UNKNOWN ? MarshalGetLastWin32Error() : Interop.Errors.ERROR_SUCCESS;
```

Other than that, looks good.